### PR TITLE
[Fix] revise aggregation impl

### PR DESCRIPTION
**Pull request by SmallSongZ**
### Task Description
revise aggregation impl

### Solution Description
add some test cases and revise aggregation impl

### Passed Regressions
N/A

### Upgrade Compatibility
N/A

### Other Information
N/A

### Release Note

N/A

### CODE REVIEW

File: src/test/java/ci.sql [hunk 1]
```diff
@@ -247,5 +247,10 @@ CREATE TABLE `test_partition_aggregation` (
     `c2` bigint DEFAULT NULL,
     PRIMARY KEY (`c1`))partition by range(`c1`)(partition p0 values less than(200), partition p1 values less than(500), partition p2 values less than(900));
 
+CREATE TABLE test_aggregation_with_filter (
```
Comment (IHEII) [inline src/test/java/ci.sql:250]: Try to use test_partition_aggregation for tests with filters.

File: src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/aggregation/ObTableAggregationResult.java [hunk 2]
```diff
@@ -43,11 +43,7 @@ public ObTableAggregationResult(QueryResultSet queryResultSet, Map<Integer, Stri
      */
     public void init() throws Exception {
         this.queryResultSet.next();
-        List<ObObj> init_row;
-        init_row = this.queryResultSet.get_Row();
-        for (int i = 0; i < init_row.size(); i++) {
-            row.put(message.get(i), init_row.get(i).getValue());
-        }
+        row = this.queryResultSet.getRow();
```
Comment (IHEII) [inline src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/aggregation/ObTableAggregationResult.java:46]: Have you ever tested ObTableAggregationResult on a non-existent raw or an empty raw?

File: src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/aggregation/ObTableAggregationResult.java [hunk 3]
```diff
@@ -26,13 +26,13 @@
 import java.util.Map;
 
 public class ObTableAggregationResult {
-    
+
     // this message is used to record the aggregation order and the corresponding aggregation name
-    private Map<Integer, String>  message; 
+    private List<String>  message;
```
Comment (IHEII) [inline src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/aggregation/ObTableAggregationResult.java:31]: I guess ``message`` is unused now.
```suggestion
```

**Review by IHEII — COMMENTED** [review]
Try to fix the review with the code cloned from your remote repo.

File: src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/aggregation/ObTableAggregationResult.java (hunk 3 above)
Comment (SmallSongZ) [reply src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/aggregation/ObTableAggregationResult.java:31]: fixed

File: src/test/java/ci.sql (hunk 1 above)
Comment (SmallSongZ) [reply src/test/java/ci.sql:250]: fixed

File: src/test/java/ci.sql [hunk 4]
```diff
@@ -238,7 +238,7 @@ CREATE TABLE test_aggregation (
     `c4` float NOT NULL,
     `c5` double NOT NULL,
     `c6` tinyint NULL,
-    `c7` date,
+    `c7` datetime,
```
Comment (IHEII) [inline src/test/java/ci.sql:241]: Why change ``date`` into ``datetime``?

File: src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/aggregation/ObTableAggregationResult.java (hunk 2 above)
Comment (SmallSongZ) [reply src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/aggregation/ObTableAggregationResult.java]: fixed

File: src/test/java/ci.sql (hunk 4 above)
Comment (SmallSongZ) [reply src/test/java/ci.sql:241]: it must use 'datetime' to adapt test case

**Comment by IHEII** [comment]
LGTM

**Pull request merged**